src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -454,18 +467,47 @@ def __str__(self):
         postgresql_server_endpoint='.postgres.database.cloudapi.de',
         mariadb_server_endpoint='.mariadb.database.cloudapi.de'))
 
-KNOWN_CLOUDS = [AZURE_PUBLIC_CLOUD, AZURE_CHINA_CLOUD, AZURE_US_GOV_CLOUD, AZURE_GERMAN_CLOUD]
 
-if 'ARM_CLOUD_METADATA_URL' in os.environ:
-    try:
-        arm_cloud_dict = json.loads(urlretrieve(os.getenv('ARM_CLOUD_METADATA_URL')))
-        cli_cloud_dict = _convert_arm_to_cli(arm_cloud_dict)
-        if 'AzureCloud' in cli_cloud_dict:
-            cli_cloud_dict['AzureCloud'].endpoints.active_directory = 'https://login.microsoftonline.com'  # change once active_directory is fixed in ARM for the public cloud
-        KNOWN_CLOUDS = list(cli_cloud_dict.values())
-    except Exception as ex:  # pylint: disable=broad-except
-        logger.warning('Failed to load cloud metadata from the url specified by ARM_CLOUD_METADATA_URL')
-        raise ex
+def get_known_clouds(refresh=False):
+    if 'ARM_CLOUD_METADATA_URL' in os.environ:
```

Comment:
It's a little confused here. Add this check in line 487 only ? Otherwise, if for some case `ARM_CLOUD_METADATA_URL` is removed from environment variables, there will be know clouds even the local cloudEndpoints.json exists.

Reply:
Again, this PR is only for improving the short-term solution. The purpose here is to not impact most users who don't care about the environment variable. So `cloudEndpoints.json` will only be filled and used when the environment variable exists.

In the long-term solution, this will change because even for public cloud, we will fetch the cloud endpoints from a url and store in `cloudEndpoints.json`. `cloudEndpoints.json` will then be loaded regardless of the environment variable's existence.